### PR TITLE
[Feature] Support using gradient checkpointing in FSDP

### PR DESCRIPTION
**Pull request by HAOCHENYE**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers. By the way, if you're not familiar with how to use pre-commit to fix lint issues or add unit tests, please refer to [Contributing to OpenMMLab](https://mmengine.readthedocs.io/en/latest/notes/contributing.html).

## Motivation

Support using gradient checkpointing in FSDP https://github.com/open-mmlab/mmdetection/pull/10981

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMPretrain.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmengine/_strategy/fsdp.py [hunk 1]
```diff
@@ -119,6 +135,12 @@ def _wrap_model(self, model: nn.Module) -> None:
             FullyShardedDataParallel: ``MMFullyShardedDataParallel``
             or subclass of ``FullyShardedDataParallel``.
         """
+        try:
+            from torch.distributed.algorithms._checkpoint.checkpoint_wrapper import \
+                apply_activation_checkpointing  # noqa: E501
```
Comment (zhouzaida) [inline mmengine/_strategy/fsdp.py:140]:
```suggestion
                apply_activation_checkpointing
```

**Review by zhouzaida — APPROVED** [review]

File: mmengine/_strategy/fsdp.py [hunk 2]
```diff
@@ -119,6 +135,12 @@ def _wrap_model(self, model: nn.Module) -> None:
             FullyShardedDataParallel: ``MMFullyShardedDataParallel``
             or subclass of ``FullyShardedDataParallel``.
         """
+        try:
+            from torch.distributed.algorithms._checkpoint.checkpoint_wrapper import \
+                apply_activation_checkpointing
```
Comment (zhouzaida) [inline mmengine/_strategy/fsdp.py:140]:
```suggestion
                apply_activation_checkpointing  # noqa: E501
```

**Review by zhouzaida — APPROVED** [review]
